GitTfs/Core/GitTfsRemote.cs

```diff
@@ -442,27 +468,102 @@ private string ProcessChangeset(ITfsChangeset changeset, LogEntry log)
 
             if (!string.IsNullOrWhiteSpace(changeset.Summary.PerformanceReviewer))
                 metadatas.Append("\nPerformance Reviewer:" + changeset.Summary.PerformanceReviewer);
+
+            if (!string.IsNullOrWhiteSpace(changeset.OmittedParentBranch))
+                metadatas.Append("\nOmitted parent branch: " + changeset.OmittedParentBranch);
+
             if (metadatas.Length != 0)
                 Repository.CreateNote(commitSha, metadatas.ToString(), log.AuthorName, log.AuthorEmail, log.Date);
             return commitSha;
         }
 
-        private string FindMergedRemoteAndFetch(int parentChangesetId, bool stopOnFailMergeCommit)
+        private string FindRootRemoteAndFetch(int parentChangesetId)
+        {
+            string omittedParentBranch;
+            return FindRemoteAndFetch(parentChangesetId, false, false, out omittedParentBranch);
+        }
+
+        private string FindMergedRemoteAndFetch(int parentChangesetId, bool stopOnFailMergeCommit, out string omittedParentBranch)
         {
-            var tfsRemotes = FindTfsRemoteOfChangeset(Tfs.GetChangeset(parentChangesetId));
-            foreach (var tfsRemote in tfsRemotes.Where(r=>string.Compare(r.TfsRepositoryPath, this.TfsRepositoryPath, StringComparison.InvariantCultureIgnoreCase) != 0))
+            return FindRemoteAndFetch(parentChangesetId, false, true, out omittedParentBranch);
```

Comment:
The `stopOnFailMergeCommit` parameter is unused and `false` is passed to the `stopOnFailMergeCommit` parameter of the called method. Should it be fixed?

Reply:
That's always good to clean the mess 😉 

Could you do a PR?

Reply:
Sure!